components/script/dom/bindings/str.rs

```diff
@@ -113,6 +113,59 @@ pub fn is_token(s: &[u8]) -> bool {
     })
 }
 
+/// Returns whether the language is matched, as defined by
+/// [RFC 4647](https://tools.ietf.org/html/rfc4647#section-3.3.2).
+pub fn extended_filtering(tag: &str, range: &str) -> bool {
+    let lang_ranges: Vec<&str> = range.split(',').collect();
+
+    lang_ranges.iter().any(|&lang_range| {
+        // step 1
+        let range_subtags: Vec<&str> = lang_range.split('\x2d').collect();
+        let tag_subtags: Vec<&str> = tag.split('\x2d').collect();
+
+        let mut range_iter = range_subtags.iter();
+        let mut tag_iter = tag_subtags.iter();
+
+        // step 2
+        // Note: [Level-4 spec](https://drafts.csswg.org/selectors/#lang-pseudo) check for wild card
+        if let (Some(range_subtag), Some(tag_subtag)) = (range_iter.next(), tag_iter.next()) {
```

Comment:
(here)[https://play.rust-lang.org/?code=fn%20main()%20%7B%0Alet%20a%20%3D%20%5B1%2C%202%2C%203%5D%3B%0A%0Alet%20mut%20iter%20%3D%20a.iter()%3B%0A%0A%2F%2F%20A%20call%20to%20next()%20returns%20the%20next%20value...%0Aassert_eq!(Some(%261)%2C%20iter.next())%3B%0Aassert_eq!(Some(%262)%2C%20iter.next())%3B%0Aassert_eq!(Some(%263)%2C%20iter.next())%3B%0A%0A%2F%2F%20...%20and%20then%20None%20once%20it%27s%20over.%0Aassert_eq!(None%2C%20iter.next())%3B%0A%0A%2F%2F%20More%20calls%20may%20or%20may%20not%20return%20None.%20Here%2C%20they%20always%20will.%0Aassert_eq!(None%2C%20iter.next())%3B%0Aassert_eq!(None%2C%20iter.next())%3B%0A%7D]

the compilation will fail if I remove `mut`